vernac/mltop.ml

```diff
@@ -339,25 +366,18 @@ type ml_module_digest =
   | NoDigest
   | AnyDigest of Digest.t list
 
-type ml_module_object = {
-  mlocal : Vernacexpr.locality_flag;
-  mnames : string list;
-  mdigests : ml_module_digest list;
-  dune_compat_logpathroot : Names.module_ident
-}
-let current_logpathroot () =
-  let open Names in
-  List.hd @@ List.rev @@ DirPath.repr @@ ModPath.dp @@ Global.current_modpath ()
+type ml_module_object =
+  { mlocal : Vernacexpr.locality_flag
+  ; mnames : PluginSpec.t list
+  ; mdigests : ml_module_digest list
+  }
 
 let cache_ml_objects mnames =
-  let use_legacy_loading_if_possible = true in
-  let iter obj = trigger_ml_object ~verbose:true ~cache:true ~reinit:true ~use_legacy_loading_if_possible obj in
+  let iter obj = trigger_ml_object ~verbose:true ~cache:true ~reinit:true obj in
   List.iter iter mnames
 
-let load_ml_objects _ {mnames; dune_compat_logpathroot; _} =
-  let use_legacy_loading_if_possible =
-    Names.Id.equal dune_compat_logpathroot (current_logpathroot ()) in
```

Comment:
I think it failed because this logic was deleted, let's see if putting it back works

Reply:
That's possible, in my testing I didn't see it was necessary, as I was covering that case by storing the legacy path in full, but it is obvious I missed a case.

Reply:
That logic was definitively bogus and not doing anything useful.